src/pykiso/lib/auxiliaries/zephyr.py

```diff
@@ -0,0 +1,223 @@
+        """Wait for the test to finish
+
+        :return: The result of the test, taken from the xunit output of twister
+        """
+        if self.process is None:
+            raise ZyphyrException("Twister was not started so can not wait.")
+
+        # Read the twister console for log output
+        while True:
+            line = self._readline()
+            if line is None:
+                break
+
+            # Look for relevant messages in the output
+            if "OUTPUT:  PASS - " in line:
+                log.info(f"Zephyr test PASSED.")
+            elif "OUTPUT:  FAIL - " in line:
+                log.info(f"Zephyr test FAILED.")
+
+        # Wait for the twister process to exit
+        ret = self.process.wait()
+
+        # Get the result from the xunit file
+        result = self._parse_xunit(str(Path(self.outdir) / "twister_report.xml"))
+
+        self.process = None
+        return result
+
+
+class ZephyrTestAuxiliary(AuxiliaryInterface):
+    """Auxiliary for Zephyr test interaction using the twister commandline tool
+
+    The functionality includes test case execution and result collection.
+
+    """
+
+    def __init__(
+        self,
+        twister_path: str = "twister",
+        test_directory: Optional[str] = None,
+        test_name: Optional[str] = None,
+        wait_for_start: bool = True,
+        **kwargs,
+    ) -> None:
+        """Initialize the auxiliary
+
+        :param twister_path: Path to the twister tool
+        :param test_directory: The directory to search for the Zephyr test project
+        :param testcase_name: The name of the Zephyr test
+        :param wait_for_start: Wait for Zyephyr test start
+
+        """
+        self.twister_path = twister_path
+        self.test_directory = test_directory
+        self.test_name = test_name
+        self.wait_for_start = wait_for_start
+        self.twister = Twister(twister_path)
+        super().__init__(auto_start=False, **kwargs)
+
+    def start_test(self, test_directory: Optional[str] = None, test_name: Optional[str] = None) -> None:
+        """Start the Zephyr test
+
+        :param test_directory: The directory to search for the Zephyr test project. Defaults to the test_directory from YAML.
+        :param testcase_name: The name of the Zephyr test. Defaults to the testcase_name from YAML.
+        """
+        test_directory = test_directory if test_directory is not None else self.test_directory
+        test_name = test_name if test_name is not None else self.test_name
+        if test_directory is None:
+            raise ZyphyrException("test_directory parameter is not set.")
+        if test_name is None:
+            raise ZyphyrException("test_name parameter is not set.")
+        return self.twister.start_test(test_directory, test_name, self.wait_for_start)
+
+    def wait_test(self) -> TestResult:
+        return self.twister.wait_test()
+
+    def _create_auxiliary_instance(self) -> bool:
```

Comment:
based on the below implementation you don't need to inherit from AuxiliaryInterface

Reply:
Derived from SimpleAuxiliaryInterface instead.